Refactor TransferData to remove canChangeValue and ignoreValueCheck

gemdev111 opened the pull request:
Eliminates the canChangeValue and ignoreValueCheck properties from TransferData, moving this logic to TransferDataType extensions. Updates all usages and initializers accordingly.
Also updates TransferDataType.transfer to include an isScanned parameter, and refactors related logic and tests to match the new structure.

closes #1010

TODO: - more tests on transfer data

gemdev111 committed:
Refactor TransferData to remove canChangeValue and ignoreValueCheck

Eliminates the canChangeValue and ignoreValueCheck properties from TransferData, moving this logic to TransferDataType extensions.
Updates all usages and initializers accordingly.
Also updates TransferDataType.transfer to include an isScanned parameter, and refactors related logic and tests to match the new structure.

gemdev111 committed:
Fix canChangeValue logic for transfer with isScanned

Updated the canChangeValue property to return false for transfer cases where isScanned is true, ensuring correct behavior when handling scanned transfers.

gemdev111 committed:
Update FeeInput initialization in Bitcoin fee tests

Added the isScanned parameter to the FeeInput type initialization in BitcoinFeeCalculatorTests to match updated FeeInput signature.

gemdev111 committed:
Update tests to use isScanned parameter in transfer mocks

Refactored test cases in ConfirmTransferViewModelTests, TransferDataViewModelTests, and SolanaFeeServiceTests to include the isScanned: false parameter when creating transfer mock objects. This ensures consistency with the updated TransferDataType API and improves test accuracy.

gemcoder21 commented on Features/Transfer/Sources/ViewModels/AmountSceneViewModel.swift:
```diff
@@ -329,14 +329,13 @@ extension AmountSceneViewModel {
         case .transfer:
             return TransferData(
-                type: .transfer(asset),
+                type: .transfer(asset, isScanned: false),
```
remove `isScanned`

DRadmir commented on Features/Staking/Sources/ViewModels/StakeDetailViewModel.swift:
```diff
@@ -134,14 +134,13 @@ public struct StakeDetailViewModel {
     
     public func withdrawStakeTransferData() throws -> TransferData {
-        TransferData(
+        return TransferData(
```
```suggestion
        TransferData(
```

DRadmir commented on Packages/Primitives/Sources/TransferDataType.swift:
```diff
@@ -7,7 +7,7 @@ public enum AccountDataType: Hashable, Equatable, Sendable {
 
 public enum TransferDataType: Hashable, Equatable, Sendable {
-    case transfer(Asset)
+    case transfer(Asset, isScanned: Bool)
```
If necessary, you can add a case for a scanned transfer

gemdev111 committed:
Refactor transfer type to use TransferMode enum

Replaces the 'isScanned' boolean in TransferDataType.transfer with a new TransferMode enum ('flexible' and 'fixed') for clearer intent and extensibility. Updates all usages, tests, and related logic to support this change, and moves value check and changeability logic into TransferDataType. Also improves TransferData id generation and testkit mocks.

gemdev111 committed:
Update TransferData+PrimitivesTests.swift

gemdev111 committed:
Merge branch 'main'

gemdev111 committed:
Refactor TransferDataType and TransferData to remove TransferMode

Removed the TransferMode enum and its usage from TransferDataType, simplifying the .transfer case to only require an Asset. The canChangeValue property is now managed directly in TransferData, with related logic updated across view models, services, and tests. This refactor streamlines transfer type handling and improves clarity in value mutability for transfer operations.

gemdev111 committed:
Merge branch 'main'

gemdev111 committed:
Add support for deposit and perpetual transfer types

Extended TransferDataType to handle deposit and perpetual types for value check logic. Updated related tests to cover new cases and refactored expectations for TransferAmountCalculator.
Removed unused ignoreValueCheck parameter in PerpetualSceneViewModel.

gemcoder21 commented on Packages/Primitives/Sources/TransferData.swift:
```diff
@@ -3,28 +3,33 @@
+            recipientData.recipient.address,
+            String(value)
+        ]
+        return components.joined(separator: "-")
```
```
[
            type.transactionType.rawValue,
            recipientData.recipient.address,
            String(value)
        ].joined(separator: "-")
```

gemdev111 committed:
Refactor TransferData id property formatting

Simplified the id property by directly joining the components array without intermediate variable assignment.
